pkg/nsx/services/vpc/vpc.go

```diff
@@ -128,6 +155,31 @@ func InitializeVPC(service common.Service) (*VPCService, error) {
 	go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeVpc, nil, VPCService.VpcStore)
 	go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeIPBlock, nil, VPCService.IpblockStore)
 
+	//initalize avi rule related store
+	if VPCService.supportAviAllowRule {
+		VPCService.RuleStore = &RuleStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.RuleBindingType(),
+		}}
+		VPCService.GroupStore = &GroupStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.GroupBindingType(),
+		}}
+		VPCService.SecurityPolicyStore = &SecurityPolicyStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.SecurityPolicyBindingType(),
+		}}
+		VPCService.PubIpblockStore = &PubIPblockStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.IpAddressBlockBindingType(),
+		}}
+		go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeGroup, nil, VPCService.GroupStore)
+		go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeRule, nil, VPCService.RuleStore)
+
+		query := fmt.Sprintf("%s:%s AND visibility:EXTERNAL", common.ResourceType, common.ResourceTypeIPBlock)
+		go VPCService.PopulateResourcetoStore(&wg, fatalErrors, common.ResourceTypeIPBlock, query, VPCService.PubIpblockStore, nil)
```

Comment:
why we don't have SecurityPolicyStore initialized here?

Reply:
securitypolicy doesn't have tags, and each vpc has only one sp, so if not found from store, it will use GET API to retrieve it.

Reply:
If we use GET API, why we need to save it into store?

Reply:
it use GET API to initialize the store, cache it to avoid more GET API invocation.